src/pytest_allure_spec_coverage/plugin.py

```diff
@@ -9,4 +9,166 @@
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import ClassVar, Iterable, List, Mapping
+
+import pytest
+from _pytest.config import Config
+from _pytest.config.argparsing import Parser
+from _pytest.config.exceptions import UsageError
+from _pytest.mark.structures import Mark
+from _pytest.nodes import Item
+from allure_commons.model2 import Label, Status, TestResult
+from allure_commons.reporter import AllureReporter
+from allure_commons.utils import uuid4
+from allure_pytest.listener import AllureListener
+from pluggy.manager import PluginManager
+
+from pytest_allure_spec_coverage.models.collector import Collector
+from pytest_allure_spec_coverage.models.scenario import Scenario
+from pytest_allure_spec_coverage.spec_collectors.sphinx import SphinxCollector
+
+
+def safe_get_marker(item: Item, name: str) -> Mark:
+    """Safely get pytest.Item marker"""
+
+    stub_marker = Mark(name="stub", args=(), kwargs={})
+    return item.get_closest_marker(name, stub_marker)
+
+
+def scenario_links(item: Item) -> Iterable[str]:
+    """Get scenario links from pytest.Item"""
+
+    return safe_get_marker(item, ScenariosMatcher.MARKER_NAME).args
+
+
+def allure_labels(scenario: Scenario) -> List[Label]:
+    """Make labels for Allure from the given scenario"""
+
+    labels = ("parentSuite", "suite", "subSuite")
+    values = [p.name for p in scenario.parents]
+    if len(values) > 3:
+        itv = slice(2, None)
+        values[itv] = [".".join(values[itv])]
+    return [Label(label, value) for label, value in zip(labels, values)]
+
+
+@dataclass
+class ScenariosMatcher:
+    """Match collected test cases with collected scenarios and report missed ones"""
+
+    PLUGIN_NAME: ClassVar[str] = "smatcher"
+    MARKER_NAME: ClassVar[str] = "scenario"
+
+    config: Config
+    collector: Collector
+    reporter: AllureReporter
+
+    scenarios: Iterable[Scenario] = field(default_factory=tuple)
+    implemented: Iterable[str] = field(default_factory=tuple)
+
+    @property
+    def missed(self) -> Iterable[Scenario]:
+        """Not implemented scenarios"""
+
+        return (scenario for scenario in self.scenarios if scenario.link not in self.implemented)
+
+    def __hash__(self) -> int:
+        return hash(self.PLUGIN_NAME)
+
+    def collect_implemented(self, items: List[pytest.Item]) -> None:
+        """Collect implemented scenario links"""
+
+        self.implemented = set(chain.from_iterable(scenario_links(item) for item in items))
+
+    def report(self) -> None:
+        """Report about not implemented scenarios"""
+
+        for scenario in self.missed:
+            self._report_scenario(scenario)
+
+    def _report_scenario(self, scenario: Scenario) -> None:
+        fake_uuid = uuid4()
+        fake_result = TestResult(
+            uuid=fake_uuid,
+            name=scenario.name,
+            description=scenario.display_name,
+            status=Status.UNKNOWN,
+            labels=allure_labels(scenario),
```

Comment:
The scenario also has `link` attr
Place `Link` allure object to `links` attr of TestResult

Reply:
Done.